mask/eolearn/mask/utils.py

```diff
@@ -38,49 +87,127 @@ def map_over_axis(data, func, axis=0):
+    else:
+        raise ValueError("Exactly one of the arguments new_size or scale_factors must be given.")
+
+    size = (width, height)
+    if resize_library is ResizeLib.CV2:
+        resize_function = partial(cv2.resize, dsize=size, interpolation=resize_method.get_cv2_method(data.dtype))
+    else:
+        resize_function = partial(_pil_resize_ndarray, size=size, method=resize_method.get_pil_method())
+
+    return _apply_to_spatial_axes(resize_function, data, spatial_axes)
+
+
+def _pil_resize_ndarray(image: np.ndarray, size: Tuple[int, int], method: Image.Resampling) -> np.ndarray:
+    return np.array(Image.fromarray(image).resize(size, method))
+
+
+def _apply_to_spatial_axes(
+    resize_function: Callable[[np.ndarray], np.ndarray], data: np.ndarray, spatial_axes: Tuple[int, int]
+) -> np.ndarray:
+    """Helper function for applying resizing to spatial axes
+
+    Recursively slices data into smaller-dimensional ones, until only the spatial axes remain. The indices of spatial
+    axes have to be adjusted if the recursion-axis is smaller than either one, e.g. spatial axes (1, 2) become (0, 1)
+    after splitting the 3D data along axis 0 into 2D arrays.
+
+    After achieving 2D data slices the resizing function is applied. The data is then reconstructed into original form.
+    """
+
+    if data.ndim <= 2:
+        return resize_function(data)
+
+    axis = next(i for i in range(data.ndim) if i not in spatial_axes)
+    data = np.moveaxis(data, axis, 0)
+
+    ax1, ax2 = (ax if axis > ax else ax - 1 for ax in spatial_axes)
 
-    return res
+    mapped_slices = [_apply_to_spatial_axes(resize_function, data_slice, (ax1, ax2)) for data_slice in data]
+    return np.moveaxis(np.stack(mapped_slices), 0, axis)
 
 
-def resize_images(data, new_size=None, scale_factors=None, anti_alias=True, interpolation="linear"):
+def resize_images(
+    data: np.ndarray,
+    new_size: Optional[Tuple[int, int]] = None,
+    scale_factors: Optional[Tuple[float, float]] = None,
+    anti_alias: bool = True,
+    interpolation: str = "linear",
+) -> np.ndarray:
     """Resizes the image(s) according to given size or scale factors.
 
     To specify the new scale use one of `new_size` or `scale_factors` parameters.
 
     :param data: input image array
-    :type data: numpy array with shape (timestamps, height, width, channels),
-                (height, width, channels), or (height, width)
     :param new_size: New size of the data (height, width)
-    :type new_size: (int, int)
     :param scale_factors: Factors (fy,fx) by which to resize the image
-    :type scale_factors: (float, float)
     :param anti_alias: Use anti aliasing smoothing operation when downsampling. Default is True.
-    :type anti_alias: bool
     :param interpolation: Interpolation method used for resampling.
                           One of 'nearest', 'linear', 'cubic'. Default is 'linear'.
-    :type interpolation: string
     """
+    warnings.warn(
```

Comment:
`resize_images` is also used by `CloudMaskTask`. So if we deprecate it we should start using `spatially_resize_image` there so that users won't be getting deprecation warnings. But then the problem will be that this one supports anti-aliasing while the new one doesn't. Hence the results of `CloudMaskTask` will change which we don't want.

I'm wondering if we shouldn't rather just join functionalities of `spatially_resize_image` and `resize_image` so that we don't have code duplications?

Reply:
I thought i forgot something...
Actually the new function does have anti-aliasing, as long as you use the PIL backend (and this is actually 3x faster than the old function). However the anti-aliasing is a bit different, so there will be some differences if I switch.